connect/resources/automation_engine.py

```diff
@@ -39,34 +42,40 @@ def process_request(self, request):
         raise NotImplementedError('Please implement `{}.process_request` method'
                                   .format(self.__class__.__name__))
 
-    @function_log(custom_logger=logger)
+    @function_log
     def approve(self, pk, data):
         # type: (str, dict) -> str
         return self._api.post(path=pk + '/approve/', json=data)[0]
 
-    @function_log(custom_logger=logger)
+    @function_log
     def inquire(self, pk):
         # type: (str) -> str
         return self._api.post(path=pk + '/inquire/', json={})[0]
 
-    @function_log(custom_logger=logger)
+    @function_log
     def fail(self, pk, reason):
         # type: (str, str) -> str
         return self._api.post(path=pk + '/fail/', json={'reason': reason})[0]
 
-    @function_log(custom_logger=logger)
+    @function_log
     def render_template(self, pk, template_id):
         # type: (str, str) -> ActivationTileResponse
         return TemplateResource(self.config).render(template_id, pk)
 
-    def _set_custom_logger(self, *args):
-        handlers = [copy.copy(hdlr) for hdlr in global_logger.handlers]
-        log_level = global_logger.level
-        self.__class__.logger.setLevel(log_level)
-        self.__class__.logger.propagate = False
-        self.__class__.logger.handlers = handlers
-        base = " %(levelname)-6s; %(asctime)s; %(name)-6s; %(module)s:%(funcName)s:line" \
-               "-%(lineno)d: %(message)s"
-        sformat = " ".join(args) + base
-        [handler.setFormatter(logging.Formatter(sformat))
-         for handler in self.__class__.logger.handlers]
+    @property
+    def logger(self):
+        # type: () -> logging.LoggerAdapter
+        request_id = self._current_request.id if self._current_request else 'global'
+        name = self.__class__.__name__ + '.' + request_id
+        if not self._logger_adapter or self._logger_adapter.logger.name != name:
+            self._logger_adapter = LoggerAdapter(logging.getLogger(name))
+        return self._logger_adapter
+
+    def _set_current_request(self, request):
+        # type: (Optional[BaseModel]) -> None
+        self._current_request = request
+        self._set_logger_prefix(request)
+
+    def _set_logger_prefix(self, request):
```

Comment:
This method will be overridden by the subclasses, providing the correct request information for each type of request.